delivery_send_to_shipper_at_operation/models/stock_picking.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_notification_sent = fields.Boolean(default=False)
+
+    def _send_confirmation_email(self):
+        for picking in self:
+            if not picking.carrier_id:
+                # If the current operation has no carrier defined, but a carrier
```

Comment:
can you move this to `_handle_delivery_notification()`?

Reply:
done